pilot/pkg/security/authz/model/generator.go

```diff
@@ -194,7 +194,7 @@ func (requestHeaderGenerator) principal(key, value string, forTCP bool) (*rbacpb
 	if err != nil {
 		return nil, err
 	}
-	m := matcher.HeaderMatcher(header, value)
+	m := matcher.HeaderMatcherWithRegex(header, value)
```

Comment:
Thanks for the PR! Can you add a release notes to the PR describing how the PR changes the current header matching behavior, and potential impacts to the users that use the current header matching? Meanwhile, the docs of current header matching behavior need to be updated and a user guide for this feature needs to be added in follow-up PRs in istio.io/.

Reply:
Thanks @lei-tang, I will update it when we reach a consensus on the solution :)

Reply:
Replacing HeaderMatcher() with HeaderMatcherWithRegex() changes the current behavior. Whether HeaderMatcherWithRegex() is used or not should be protected by a feature-enablement flag.

Reply:
a feature-enablement flag sgtm.